### ISSUE TITLE
Where clauses can be empty

### ISSUE THREAD

**Issue by brauliobz** [intellij-rust/intellij-rust#1671]
The code below shouldn't generate syntax errors. That's only for consistency with the language, though. In my opinion it doesn't make sense to allow this.

```rust
fn empty_generic_params<>() where {}
trait EmptyGenericParams<> where {}
trait TraitItemEmptyGenericParams where {
    fn f<>() where ;
    fn g<>(&self) where ;
}
impl <> EmptyGenericParams for i32 where {}
impl TraitItemEmptyGenericParams for i32 where {
    fn f<>() where {}
    fn g<>(&self) where {}
}
struct A {}
impl A {
    fn f<>() where {}
    fn g<>(&self) where {}
}
struct Struct<> where {}
```
